isthmus/src/main/java/io/substrait/isthmus/SqlConverterBase.java

```diff
@@ -103,6 +110,39 @@ Pair<SqlValidator, CalciteCatalogReader> registerCreateTables(List<String> table
     return Pair.of(validator, catalogReader);
   }
 
+  Result registerCreateTablesForExtendedExpression(List<String> tables) throws SqlParseException {
+    Map<String, RelDataType> nameToTypeMap = new LinkedHashMap<>();
+    Map<String, RexNode> nameToNodeMap = new HashMap<>();
+    CalciteSchema rootSchema = CalciteSchema.createRootSchema(false);
+    CalciteCatalogReader catalogReader =
+        new CalciteCatalogReader(rootSchema, List.of(), factory, config);
+    SqlValidator validator = Validator.create(factory, catalogReader, SqlValidator.Config.DEFAULT);
+    if (tables != null) {
+      for (String tableDef : tables) {
+        List<DefinedTable> tList = parseCreateTable(factory, validator, tableDef);
+        for (DefinedTable t : tList) {
+          rootSchema.add(t.getName(), t);
+          for (RelDataTypeField field : t.type.getFieldList()) {
+            nameToTypeMap.put(
+                field.getName(), field.getType()); // to validate the sql expression tree
```

Comment:
Does this handle multiple tables containing a field with the same name?

Reply:
I'll test this scenario